Bottom: 9e12ec9e014127ce37fdda36f87a03d33b4756a0
Top:    e1d96a3ae963f6ee6de4725161091b05012768a0
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-05-14 01:12:39 +0100

Refresh of gtk-gl-refactor-gl-colour-setu

---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 10ecbc2..409c182 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -45,6 +45,7 @@ typedef struct render_priv {
   bool trans_lines;
   bool in_context;
   int subcomposite_stencil_bit;
+  char *current_colorname;
 } render_priv;
 
 
@@ -351,21 +352,20 @@ set_gl_color_for_gc (hidGC gc)
 {
   render_priv *priv = gport->render_priv;
   static void *cache = NULL;
-  static char *old_name = NULL;
   hidval cval;
   ColorCache *cc;
   double alpha_mult = 1.0;
   double r, g, b, a;
   a = 1.0;
 
-  if (old_name != NULL)
-    {
-      if (strcmp (gc->colorname, old_name) == 0)
-        return;
-      free (old_name);
-    }
+  if (priv->current_colorname != NULL &&
+      strcmp (priv->current_colorname, gc->colorname) == 0)
+    return;
+
+  free (priv->current_colorname);
+  priv->current_colorname = NULL;
 
-  old_name = strdup (gc->colorname);
+  priv->current_colorname = strdup (gc->colorname);
 
   if (gport->colormap == NULL)
     gport->colormap = gtk_widget_get_colormap (gport->top_window);
